pymc/logprob/mixture.py

```diff
@@ -289,7 +289,13 @@ def find_measurable_index_mixture(fgraph, node):
         # We don't support (non-scalar) integer array indexing as it can pick repeated values,
         # but the Mixture logprob assumes all mixture values are independent
         if any(
-            indices.dtype.startswith("int") and sum(1 - b for b in indices.type.broadcastable) > 0
+            (
+                isinstance(indices, (type(NoneConst) | type(None)))
```

Comment:
I think this is what you need more directly: `isinstance(indices.type, NoneTypeT)`